File: CivitasRuby/lib/civitas/casilla_sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaSorpresa < Casilla
    
    def initialize(mazo, nombre)
      super(nombre, nil, -1, -1, mazo, nil)
    end
    
    def recibeJugador(actual, todos)
      if (jugadorCorrecto(actual, todos))
        informe(actual, todos)
        @sorpresa = @mazo.siguiente
        @sorpresa.aplicarAJugador(actual, todos)
      end
    end
    
    def informe (actual, todos)
      str = "El jugador #{todos[actual].nombre} ha caido en una casilla de SORPRESA"
      Diario.instance.ocurre_evento(str)
    end
    
  end
end

File: P5/civitasJava/src/civitas/CasillaSorpresa.java
/**
 * @file CasillaImpuesto.java
 * @author Yesenia González Dávila
 * @author Esther García Gallego
 * @brief Casilla de tipo sorpresa. Subclase de Casilla.
 * @note Grupo B.3
*/

package civitas;

import java.util.ArrayList;

public class CasillaSorpresa extends Casilla {
    
    /** Constructor sorpresa. */
    CasillaSorpresa (String nombre, MazoSorpresas mazo) {
        super(nombre);
        this.mazo = mazo;
        tipoCasilla = "SORPRESA";
    }
    
    /** Recibe al jugador en la casilla, aplicando la sorpresa
     * correspondiente segun el mazo. */
    @Override
    void recibeJugador (int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto (actual, todos)) {
            sorpresa = mazo.siguiente();
            informe (actual, todos);
            sorpresa.aplicarAJugador(actual, todos);
        }
    }
}
